src/main/java/com/tozny/sdk/RealmApi.java

```diff
@@ -247,10 +247,11 @@ private boolean dispatchUserExists(UserExistsRequest req) throws ToznyApiExcepti
     }
 
     private boolean dispatchUserPush(UserPushRequest request) throws ToznyApiException {
-        ToznyApiResponse<Boolean> response = protocol.<ToznyApiResponse<Boolean>>dispatch(
-                request, new TypeReference<ToznyApiResponse<Boolean>>() {});
-        String ret = response.getReturn();
-        return ret != null && ret.equals("true");
+        ToznyApiResponse<Boolean[]> response = protocol.<ToznyApiResponse<Boolean[]>>dispatch(
```

Comment:
Yes. The literal JSON returned from the API is either:

```json
{"return": "ok", "status_code": 200, "results": [false], "total": 1}
```

Or

```json
{"return": "ok", "status_code": 200, "results": [true], "total": 1}
```

We're looking for the contents of the `results` array.